metrics_activities.go

```diff
@@ -1,39 +1,40 @@
 package chartmogul
 
-import "strings"
-
 // MetricsActivity represents Metrics API activity in ChartMogul.
 type MetricsActivity struct {
-	ID                  uint64  `json:"id"`
-	Date                string  `json:"date"`
-	ActivityArr         float64 `json:"activity-arr"`
-	ActivityMrr         float64 `json:"activity-mrr"`
-	ActivityMrrMovement float64 `json:"activity-mrr-movement"`
-	Currency            string  `json:"currency"`
-	CurrencySign        string  `json:"currency-sign"`
-	Description         string  `json:"description"`
-	Type                string  `json:"type"`
+	Date                    string  `json:"date"`
+	ActivityArr             float64 `json:"activity-arr"`
+	ActivityMrr             float64 `json:"activity-mrr"`
+	ActivityMrrMovement     float64 `json:"activity-mrr-movement"`
+	Currency                string  `json:"currency"`
+	Description             string  `json:"description"`
+	Type                    string  `json:"type"`
+	SubscriptionExternalID  string  `json:"subscription-external-id"`
+	PlanExternalID          string  `json:"plan-external-id"`
+	CustomerName            string  `json:"customer-name"`
+	CustomerUUID            string  `json:"customer-uuid"`
+	CustomerExternalID      string  `json:"customer-external-id"`
+	BillingConnectorUUID    string  `json:"billing-connector-uuid"`
+	UUID                    string  `json:"uuid"`
 }
 
 // MetricsActivities is the result of listing activities in Metrics API.
 type MetricsActivities struct {
 	Entries []*MetricsActivity `json:"entries"`
 	HasMore bool               `json:"has_more"`
 	PerPage uint32             `json:"per_page"`
-	Page    uint32             `json:"page"`
```

Comment:
why did we remove it from response?

Reply:
This is actually the new v1/activities endpoint which does not use the offset based pagination. It instead uses an anchor record based pagination strategy. The user is expected to send back the last uuid of the last response to get more records.
The existing endpoint is available at MetricsListCustomerActivities and continues to use the standard offset based pagination with page param.